lib/revision_table_access_check_filter.js

```diff
@@ -81,6 +81,13 @@ module.exports = function(hyper, req, next, options, specInfo) {
                 }
             });
         }
-        return next(hyper, req);
+        return next(hyper, req)
+        .then(function(res) {
+            if (req.query.redirect === false || mwUtil.isNoCacheRequest(req)) {
+                res.status = 200;
+                delete res.headers.location;
```

Comment:
[Varnish removes it](https://github.com/wikimedia/operations-puppet/blob/c53f1476ab6cddad6150ce85f1441646ba92444b/modules/varnish/templates/text-frontend.inc.vcl.erb#L232-L235), and so should we.